pkg/binder/framework/plugins/podtopologyspread/podtopologyspread.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2023 The Godel Scheduler Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podtopologyspread
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kubewharf/godel-scheduler/pkg/binder/framework/handle"
+	framework "github.com/kubewharf/godel-scheduler/pkg/framework/api"
+	"github.com/kubewharf/godel-scheduler/pkg/plugins/helper"
+	nodeInfoHelper "github.com/kubewharf/godel-scheduler/pkg/plugins/helper"
+	"github.com/kubewharf/godel-scheduler/pkg/plugins/podlauncher"
+	utils "github.com/kubewharf/godel-scheduler/pkg/plugins/podtopologyspread"
+	"github.com/kubewharf/godel-scheduler/pkg/scheduler/apis/config"
+	"github.com/kubewharf/godel-scheduler/pkg/scheduler/apis/validation"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	Name = "PodTopologySpreadCheck" //Name
+)
+
+type TopologySpreadCondition struct {
+	Constraints          []utils.TopologySpreadConstraint
+	TpKeyToCriticalPaths map[string]*utils.CriticalPaths
+	TpPairToMatchNum     map[utils.TopologyPair]*int32
+}
+
+type PodTopologySpreadCheck struct {
+	args            config.PodTopologySpreadArgs
+	frameworkHandle handle.BinderFrameworkHandle
+}
+
+var _ framework.CheckTopologyPlugin = &PodTopologySpreadCheck{}
+
+func (pl *PodTopologySpreadCheck) Name() string {
+	return Name
+}
+
+func (pl *PodTopologySpreadCheck) CheckTopology(_ context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeInfo framework.NodeInfo) *framework.Status {
+	podLauncher, status := podlauncher.NodeFits(cycleState, pod, nodeInfo)
+	if status != nil {
+		return status
+	}
+
+	constraints, err := pl.getConstraints(pod)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+	if len(constraints) == 0 {
+		return nil
+	}
+
+	nodeInfos, err := nodeInfoHelper.ReadAllNodeInfos(cycleState)
+	if err != nil {
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
+	state := utils.GetPreFilterState(pod, nodeInfos, constraints)
```

Comment:
There is an uncommon case: the pod can be scheduled only after preempting the victims. In this case, the removal of the victims should be awared in the state, just as the `AddPod` and `RemovePod` functions of InterPodAffininity/PodTopologySpread plugins do.